### PR TITLE
avoid mixed usage of Array<T, N> and T[N] when passing to template function

### PR DESCRIPTION
**Pull request by liqiangxl**
Issue: https://github.com/NVIDIA/Fuser/issues/1470
In the generated code of the case posted in the issue page:
`nvfuser_index_t T5[4]` is aliased as `Array<float, 4> T9.` `float T4[4]` is aliased as `auto& T10 = T4`.
Then T9 and T10 are used to call `welfordGroupOuter` and causes a compilation error due to type mismatch: T9 is a custom array type, while T10 is a native float array.

Fix: always pass the regular array of registers to template function.

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build --diff --diff-bench

**Comment by liqiangxl** [comment]
!build --diff --diff-bench

**Comment by liqiangxl** [comment]
codediff in matmul cases are caused by `Cleanup matmul test utilities (#1804)`

File: csrc/codegen.cpp [hunk 1]
```diff
@@ -246,6 +249,24 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     return val_to_name_.at(v);
   }
 
+  // If the variable is an aligned array, append ".array" to use the reguar
+  // array. This avoid the type mismatch in template functions when one of the
+  // arguments is an aligned array (Array<T,N>) while the other is a regular
+  // array T[N].
+  std::string genVarForTemplateFunction(Val* v) {
```
Comment (zasdfgbnm) [inline csrc/codegen.cpp:256]: Should this be named something like `maybeConvertAlignedArrayToRawArray`?

**Review by zasdfgbnm — APPROVED** [review]

File: csrc/codegen.cpp (hunk 1 above)
Comment (liqiangxl) [reply csrc/codegen.cpp:256]: I rename to `genVariableNameConvertAlignedArray` to reflect its two functions: (1) call `genVariableName` and (2) ConvertAlignedArrayToRawArray

**Pull request merged**